### DISCUSSION TITLE
Nullable reference types

### DISCUSSION THREAD

**Discussion by henda79** [Azure/azure-mobile-apps#675] [General]
I'm using nullable reference types as it helps with NullReferenceException at runtime.

```
  public class SynchronizationEventArgs
  {
    /// <summary>The type of event.</summary>
    public SynchronizationEventType EventType { get; internal set; }

    /// <summary>
    /// When an item is indicated, the ID of the item that was processed.
    /// </summary>
    public string ItemId { get; internal set; }

    /// <summary>
    /// When pulling records, the number of items that have been processed.
    /// </summary>
    public long ItemsProcessed { get; internal set; } = -1;

    /// <summary>The number of items remaining in the queue.</summary>
    public long QueueLength { get; internal set; }

    /// <summary>
    /// The name of the table containing the item (if appropriate).
    /// </summary>
    public string TableName { get; internal set; }

    /// <summary>
    /// In the case of a result, whether the push was successful or not.
    /// </summary>
    public bool? IsSuccessful { get; internal set; }
  }
```

When I use these properties, I know from debug that some of these can return null such as TableName but since they are not marked with `?` using code like

```
if (args.TableName.Equals(...))
```

will not show any warning but will throw at runtime.

Anyway, I'm sure you know what I mean... hope you're well Adrian !

**Comment by adrianhall** [comment]
I know what you mean, and generally agree that we should set nullable enable in the library projects.  That is, however, a HUGE change in the library.  Explicitly, it is a breaking change for absolutely everyone, which is why I have not prioritized it before the v6 release.

**Reply by henda79** [reply]
I figured that would be the case. I'm working on a small new MAUI project and have enabled it from the start and is a great help.

Speaking of v6, is there a roadmap for this. I've not been following it closely as the long term project on working on is Xamarin Forms and will still require version 5, but I'm keen to have a look at version 6 for this small MAUI project.

**Reply by adrianhall** [reply]
Since I've had very little feedback on my v6 discussion threads, I've pretty much shelved what I was going to do.  I look at it now and then, but there doesn't seem to be very much need for additional work. 

My focus instead has been on how to make the offline store more self-contained in the v5 context.  This means, for example, pulling the delta token store into the main sqlite store.  I have some other ideas here, but it's been pretty quiet recently on input from the users.

**Reply by henda79** [reply]
That's a shame, maybe others are just spectating from the side lines.

What is the meaning of "delta token store" ?

**Reply by adrianhall** [reply]
There are three additional "system tables" that are included in the offline store: The "delta token store" is a store for the last synchronization time for each table; the "operations queue" is a list of changes that are waiting to be pushed; and the "sync errors" is the list of unrecoverable errors that were not handled during a synchronization call.
